adr/0001-stac-admin-plugin-system.md

```diff
@@ -0,0 +1,37 @@
+# STAC admin plugin system
+
+    Status: IN REVIEW
+    Deciders: @danielfdsilva @emmanuelmathot @j08lue @oliverroick
+    Date: 2024-08-26
+
+
+## Context and Problem Statement
+
+The form to edit the STAC metadata may differ from instance to instance and from collection to collection. Ideally the form should be easily customizable and extensible to suit the different needs.
+
+## Decision Drivers
+
+The chosen option should allow:
+- Each instance to define what fields the form editor should have;
+- Within an instance, each collection/item to define what fields the form editor should have based on some sort of, possibly dynamic, criteria;
+
+## Decision
+
+The structure of the form is defined by a plugin system. Each plugin is responsible for a section of the editor and defines the fields that should be shown.
```

Comment:
Will there be a plugin for each STAC extension to support (easier to identify and maintain) or could a plugin be in charge of a set of extensions ?

Reply:
I thinks that ideally is one plugin per extension, but this is not a rigid requirement.
And even if we do a plugin per extensions, we could then bundle plugins together in a way they make sense, like "common extensions".